### ISSUE TITLE
Keyword arugments in `@model` are not treated as data

### ISSUE THREAD

**Issue by xukai92** [TuringLang/Turing.jl#911]
In the MEW below, `x_inf` is not treated as data.

```julia
using Turing

@model mwe(x0=1.0) = begin
    m ~ Normal(0, 1)
    x0 ~ Normal(m, 1)
end

sample(mwe(), HMC(0.1, 2), 100)
```

However, this works

```julia
sample(mwe(1.0), HMC(0.1, 2), 100)
```

**Comment by xukai92** [comment]
Reported by @jonasmac16.

**Comment by mohamed82008** [comment]
This is the missing data support (https://turing.ml/dev/docs/using-turing/guide#using-argument-defaults). It's un-intuitive though I can see. This was done for handling missing data in the vector case, where the "default value" is used to find the length and type of the vector. I am not happy about this part of the API, so if we find an alternative I am happy to implement it.

**Comment by mohamed82008** [comment]
Closed via #965.
